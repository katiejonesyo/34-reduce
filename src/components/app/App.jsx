import React, { useReducer } from 'react';

const initialState = {
  color: 'green',
  before: [],
  after: []
};

function App() {
  const { current, undo, redo, record } = useRecord('#b2ff66');

  return (
    <>
      <button data-testid="undo" onClick={undo}>undo</button>
      <button data-testid="redo" onClick={redo}>redo</button>
      <label htmlFor="color-picker">color-picker</label>
      <input
        id="color-picker"
        type="color"
        value={current}
        onChange={({ target }) => record(target.value)}/>
      <div
        data-testid="color-box"
        style={{ backgroundColor: current, width: '10rem', height: '10rem' }}
      ></div>
    </>
  );
}	
export default App;



