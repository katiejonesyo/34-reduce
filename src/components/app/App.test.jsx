import React from 'react';
import {render, cleanup, screen, fireEvent, waitFor } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('sets the color to green', async() => {
    render(<App/>);
    const colorPicker = await screen.findByLabelText('color-picker');
    const colorBox = await screen.findByTestId('color-box');

    fireEvent.change(colorPicker, {
      target: {
        value: '#b2ff66'
      }
    });

    return waitFor(() => {
      expect(colorBox.style.backgroundColor).toBe('rgb(178, 255, 102)');
    });
  });

  it('Undo previous color', async() => {
    render(<App />);
    const colorPicker = await screen.findByLabelText('color-picker');
    const colorBox = await screen.findByTestId('color-box');
    const undo =  await screen.findByTestId('undo');

    fireEvent.change(colorPicker, {
      target: {
        value: '#b2ff68'
      }
    });
    fireEvent.click(undo);

    return waitFor(() => {
      expect(colorBox.style.backgroundColor).toBe('rgb(178, 255, 102)');
    });
  });

  it('redo setting the color to green', async() => {
    render(<App/>);

    const colorPicker = await screen.findByLabelText('color-picker');
    const colorBox = await screen.findByTestId('color-box');
    const undo = await screen.findByText('undo');
    const redo = await screen.findByText('redo');

    fireEvent.change(colorPicker, {
      target: {
        value: '#b2ff66'
      }
    });

    fireEvent.click(undo);
    fireEvent.click(redo);

    return waitFor(() => {
      expect(colorBox.style.backgroundColor).toBe('rgb(178, 255, 102)');
    });
  });
});



//EXTRA LINES FOR DAN



