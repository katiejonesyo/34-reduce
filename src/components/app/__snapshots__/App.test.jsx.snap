// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <button
    data-testid="undo"
    id="COLOR_UNDO"
  >
    undo
  </button>
  <button
    data-testid="redo"
    id="COLOR_REDO"
  >
    redo
  </button>
  <label
    for="COLOR_CHANGE"
  >
    color-picker
  </label>
  <input
    id="COLOR_CHANGE"
    type="color"
    value="#b2ff66"
  />
  <div
    data-testid="color-box"
    style="background-color: rgb(178, 255, 102); width: 10rem; height: 10rem;"
  />
</DocumentFragment>
`;
